build_tools/teatime.py

```diff
@@ -95,6 +176,25 @@ def finish(self):
                     f"END\t{end_time}\t{end_time - self.start_time}\n".encode()
                 )
             self.log_file.close()
+
+            # Call log indexing before upload
+            amdgpu_families = os.getenv("AMDGPU_FAMILIES", "unknown")
+            indexer_script = Path("build_tools/index_logs.sh")
+            indexer_args = [
+                str(indexer_script),
+                amdgpu_families,
+                str(self.log_path.parent),
+                "build/indexer.py",
+            ]
+            if indexer_script.exists():
+                try:
+                    print(f"[teatime] Running {indexer_script} for {amdgpu_families}")
```

Comment:
I don't think this generic script should know anything about AMDGPU_FAMILIES or other project-specific options. May need to keep index generation in other scripts.

Reply:
Any suggestion for the location -  how about I can write short script that is called by ` teatime.py`? So this will keep away the details from `teatime.py`.